File: C++/프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int> pq;
    
    for(auto& sco : scoville) {
        pq.push(-sco);
    }
    if(K <= -pq.top()) {
        return answer;
    }
    
    int sco1, sco2;
    while (2 <= pq.size()) {
        answer++;
        sco1 = -pq.top(); pq.pop();
        sco2 = -pq.top(); pq.pop();
        
        pq.push(-(sco1 + sco2 * 2));
        if(K <= -pq.top()) {
            return answer;
        }
    }
    
    return -1;
}

File: Python3/프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while 2 <= len(scoville):
        if all(i >= K for i in scoville):
            return answer
        
        answer += 1
        heapq.heappush(scoville, heapq.heappop(scoville) + (heapq.heappop(scoville) * 2))
        
    if all(i >= K for i in scoville):
        return answer
    return -1